examples/js/loaders/GLTF2Loader.js

```diff
@@ -2312,6 +2464,47 @@ THREE.GLTF2Loader = ( function () {
 
 					}
 
+					// for Specular-Glossiness
+					if ( child.material && child.material.type === 'ShaderMaterial' ) {
+
+						child.onBeforeRender = function ( renderer, scene, camera, geometry, material, group ) {
+
+							var uniforms = material.uniforms;
+							var defines = material.defines;
+
+							uniforms.diffuse.value.copy( material.color );
+							uniforms.emissive.value.copy( material.emissive );
+							uniforms.specular.value.copy( material.specular );
+							uniforms.glossiness.value = material.glossiness;
+
+							uniforms.map.value = material.map;
+							uniforms.aoMap.value = material.aoMap;
+							uniforms.bumpMap.value = material.bumpMap;
+							uniforms.emissiveMap.value = material.emissiveMap;
+							uniforms.envMap.value = material.envMap;
+							uniforms.normalMap.value = material.normalMap;
+							uniforms.specularMap.value = material.specularMap;
+							uniforms.glossinessMap.value = material.glossinessMap;
```

Comment:
Yes, this is a pain, but it is illustrative to show how it is done...

It is also why I prefer the oddly-named `MeshStandardMaterialSG` as a built-in material over this `ShaderMaterial` approach.